src/ray/core_worker/object_interface.cc

```diff
@@ -80,24 +94,107 @@ Status CoreWorkerObjectInterface::Get(
         ids, timeout_ms, worker_context_.GetCurrentTaskID(), &objects));
     RAY_CHECK(ids.size() == objects.size());
     for (size_t i = 0; i < objects.size(); i++) {
-      (*results).emplace(ids[i], objects[i]);
+      if (objects[i] != nullptr) {
+        (*results).emplace(ids[i], objects[i]);
+      }
+    }
+  }
+
+  return Status::OK();
+}
+
+Status CoreWorkerObjectInterface::Wait(const std::vector<ObjectID> &ids, int num_objects,
+                                       int64_t timeout_ms, std::vector<bool> *results) {
+  (*results).resize(ids.size(), false);
+
+  if (num_objects <= 0 || num_objects > ids.size()) {
+    return Status::Invalid("num_objects value is not valid");
+  }
+
+  // There can be a few cases here:
```

Comment:
Same as above, move to `GetObjectIdsPerStoreProvider` header

Reply:
yep, thanks